chain/blocksync/client.go

```diff
@@ -308,6 +308,12 @@ func (client *BlockSync) GetChainMessages(
 	length uint64,
 ) ([]*CompactedMessages, error) {
 	ctx, span := trace.StartSpan(ctx, "GetChainMessages")
+	if span.IsRecordingEvents() {
```

Comment:
You probably want to track this in `doRequest` which is now the single point of entry for the many APIs exposed by blocksync.

Reply:
Doing it this way means that the lengths are attached to the top level requests (and can be easily distinguished when looking at traces)